### DISCUSSION TITLE
Open modal on row click?

### DISCUSSION THREAD

**Discussion by kvnsya** [rappasoft/laravel-livewire-tables#1250] [Q&A]
Hey is there any way i can trigger a modal if i click a row.
I would like to click a row and give extra info off the clicked row in a modal.  
( Like click an order and give all products off that order in a modal )

I only find the function ->setTableRowUrl()
Or can i trigger a modal from a LinkColumn? 

I see that there are options in v1 but i can not find it in v2

Thnx,

**Comment by lrljoe** [comment]
You can set the row click events.

How are you adding your modal to the component or are you emitting to another component?

**Comment by kvnsya** [comment]
Thanks for the reply @lrljoe 

I made it work like this. 
Is there a better way?

trigger modal on row click and other button

![image](https://github.com/rappasoft/laravel-livewire-tables/assets/129859397/d7baceb4-b65a-43a8-b2f9-957e180ef591)
![image](https://github.com/rappasoft/laravel-livewire-tables/assets/129859397/a4a39489-9131-4b91-a874-048d2f5612f8)

Functions to load modal open and close

![image](https://github.com/rappasoft/laravel-livewire-tables/assets/129859397/7946181d-8201-48b6-a491-96fa2a2ae377)

My modal 

![image](https://github.com/rappasoft/laravel-livewire-tables/assets/129859397/7cd30be5-aa4a-4105-a62b-c1511f023eea)

**Reply by lrljoe** [reply]
So the only real issue with what you're doing is that you'll reload the table component, and re-execute the query as part of the render call.

If you're just loading some extra related data and displaying it in a modal, then it may be easier to just load it with the table query, and populate the modal that way.

If you're looking to manipulate that data however (e.g. edit) then  I'd highly recommend checking out the WireElements Modal approach, as you can separate out your logic for working with the modal data.
